util/pkg/vfs/context.go

```diff
@@ -309,3 +313,21 @@ func (c *VFSContext) getGCSClient() (*storage.Service, error) {
 	c.gcsClient = gcsClient
 	return gcsClient, nil
 }
+
+func (c *VFSContext) buildOpenstackSwiftPath(p string) (*SwiftPath, error) {
+	u, err := url.Parse(p)
+	if err != nil {
+		return nil, fmt.Errorf("invalid openstack cloud storage path: %q", p)
+	}
+
+	if u.Scheme != "swift" {
+		return nil, fmt.Errorf("invalid openstack cloud storage path: %q", p)
+	}
+
+	swiftClient, err := NewSwiftClient()
```

Comment:
In future, you might want to cache this, for 2 reasons:

1) efficiency in reusing the swift client
2) it means we can configure the swift client specially, either for testing purposes or because we want to pass special options to it

But we can definitely wait until it is needed!

Reply:
good comments! I understand and will update it!